### PR TITLE
PLT-3535 MarloweBulkSync Server

### PR DESCRIPTION
**Pull request by jhbertra**
- [x] Added new bulk sync protocol.
- [x] Added a server implementation to `marlowe-sync`
- [x] Rewrote marlowe-finder to use bulk sync.

Pre-submit checklist:
- Branch
    - [x] Tests are provided (if possible)
    - [ ] [Test report is updated](https://github.com/input-output-hk/marlowe-cardano/blob/main/marlowe/test/test-report.md) (if relevant)
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [ ] Relevant tickets are mentioned in commit messages
    - [x] Formatting, PNG optimization, etc. are updated
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
        - Review required
        - [x] Substantial changes to code, test, or documentation
        - [ ] Change made to Marlowe validator (@bwbush and @palas must be included as reviewers)
        - Review not required
        - [ ] Minor changes to non-critical code, documentation, nix derivations, configuration files, or scripts
        - [ ] Formatting, spelling, grammar, or reorganization
    - [x] Reviewer requested

### CODE REVIEW

File: marlowe-runtime/config/Language/Marlowe/Runtime/CLI/Option.hs [hunk 1]
```diff
@@ -49,6 +49,14 @@ syncHeaderPort =
     3725
     "The port number of the marlowe-sync server's header synchronization API."
 
+syncBulkPort :: CliOption OptionFields PortNumber
+syncBulkPort =
+  port
+    "marlowe-bulk"
+    "SYNC_MARLOWE_BULK"
+    3730
+    "The port number of the marlowe-sync server's bulk synchronization API."
```
Comment (bwbush) [inline marlowe-runtime/config/Language/Marlowe/Runtime/CLI/Option.hs:58]: Won't the operables, deployment, and compose derivations need updating, too?

In fact, should we add this reminder to the PR template?

Comment (jhbertra) [reply marlowe-runtime/config/Language/Marlowe/Runtime/CLI/Option.hs:58]: That might be a good way to deal with this. You're right we often forget this. I can't really think of a good way to automate this check.

**Review by bwbush — DISMISSED** [review]
It's great to finally have this API.

However, `marlowe-finder` does not report about 30% of the contracts on `preprod`:
```
$ marlowe-finder --end-at-tip \
| jq -r 'select(.NewContract) | .NewContract.fields.contractId \
| (.txId + "#" + (.txIx | tostring))' \
| wc -l
5123

$ psql preprod
preprod=> select count(*) from marlowe.createtxout;
 count 
-------
  7055
(1 row)
```

**Comment by jhbertra** [comment]
Thanks for performing that test, I'll investigate!

**Review by bwbush — APPROVED** [review]
@jhbertra, I reran the tests on the new commit and results are correct both on `preprod` and `preview`.

**Pull request merged**